lib/command.js

```diff
@@ -131,47 +131,62 @@ module.exports = function (yargs, usage, validation) {
   }
 
   self.runCommand = function (command, yargs, parsed) {
-    var argv = parsed.argv
+    var aliases = parsed.aliases
     var commandHandler = handlers[command] || handlers[aliasMap[command]]
-    var innerArgv = argv
     var currentContext = yargs.getContext()
     var numFiles = currentContext.files.length
     var parentCommands = currentContext.commands.slice()
+
+    // what does yargs look like after the buidler is run?
+    var innerArgv = parsed.argv
+    var innerYargs = null
+
     currentContext.commands.push(command)
     if (typeof commandHandler.builder === 'function') {
       // a function can be provided, which builds
       // up a yargs chain and possibly returns it.
-      innerArgv = commandHandler.builder(yargs.reset(parsed.aliases))
+      innerYargs = commandHandler.builder(yargs.reset(parsed.aliases))
       // if the builder function did not yet parse argv with reset yargs
       // and did not explicitly set a usage() string, then apply the
       // original command string as usage() for consistent behavior with
-      // options object below
+      // options object below.
       if (yargs.parsed === false) {
         if (typeof yargs.getUsageInstance().getUsage() === 'undefined') {
           yargs.usage('$0 ' + (parentCommands.length ? parentCommands.join(' ') + ' ' : '') + commandHandler.original)
         }
-        innerArgv = innerArgv ? innerArgv.argv : yargs.argv
+        innerArgv = innerYargs ? innerYargs._parseArgs(null, null, true) : yargs._parseArgs(null, null, true)
       } else {
         innerArgv = yargs.parsed.argv
       }
+
+      if (innerYargs && yargs.parsed === false) aliases = innerYargs.parsed.aliases
+      else aliases = yargs.parsed.aliases
     } else if (typeof commandHandler.builder === 'object') {
       // as a short hand, an object can instead be provided, specifying
       // the options that a command takes.
-      innerArgv = yargs.reset(parsed.aliases)
-      innerArgv.usage('$0 ' + (parentCommands.length ? parentCommands.join(' ') + ' ' : '') + commandHandler.original)
+      innerYargs = yargs.reset(parsed.aliases)
+      innerYargs.usage('$0 ' + (parentCommands.length ? parentCommands.join(' ') + ' ' : '') + commandHandler.original)
       Object.keys(commandHandler.builder).forEach(function (key) {
-        innerArgv.option(key, commandHandler.builder[key])
+        innerYargs.option(key, commandHandler.builder[key])
       })
-      innerArgv = innerArgv.argv
+      innerArgv = innerYargs._parseArgs(null, null, true)
+      aliases = innerYargs.parsed.aliases
     }
+
     if (!yargs._hasOutput()) populatePositionals(commandHandler, innerArgv, currentContext, yargs)
 
     if (commandHandler.handler && !yargs._hasOutput()) {
       commandHandler.handler(innerArgv)
     }
+
+    // we apply validation post-hoc, to so that custom
+    // checks get passed populated positional arguments.
+    yargs._runValidation(innerArgv, aliases)
```

Comment:
the little bit of refactoring in this method was to hold off on applying validation until after we've populated variadic values -- this was largely so that one can provide a `.check()` at the top level, and have it passed the appropriate post-processed object after commands are parsed.